src/network/constants.rs

```diff
@@ -42,7 +42,7 @@ use std::{fmt, io, ops};
 use consensus::encode::{self, Encodable, Decodable};
 
 /// Version of the protocol as appearing in network message headers
-pub const PROTOCOL_VERSION: u32 = 70001;
+pub const PROTOCOL_VERSION: u32 = 70016;
```

Comment:
If you don't signal wtxid-relay version, your peer won't start to talk wtxid-relay with you. But if we upgrade PROTOCOL_VERSION, downstream software need to implement wtxid-relay to avoid interferences while talking with Core. Once more we should copy undocumented const from upstream (src/version.h) and implementation should set PROTOCOL_VERSION to what features set they support.

Reply:
Can we add some docs here about this number? I thought it was a result of ORing flags, but I see that also in bitcoin core it's hard coded like that.

Reply:
Good point about forcing downstream to implement it, I'll change it back to 70001.
What sort of docs were you thinking? Telling people to change it to reflect the features they support or how it's derived?

Reply:
Both :)

Reply:
> But if we upgrade PROTOCOL_VERSION, downstream software need to implement wtxid-relay to avoid interferences while talking with Core

I don't think this is true? If you don't send the "wtxidrelay" message the only change is that you'll receive a new unknown message type, which is fine. IIUC, the only reason the protocol version was bumped here was that the message happens between version and verack and some folks may fail the connection for such messages.

Reply:
Right, I meant actually signaling PROTOCOL_VERSION >= WTXID_RELAY_VERSION _and_ sending `wtxidrelay` _and_ not effectively implementing wtxid-relay would provoke interference. This should be documented with at least a reference to BIP339 specifications.

In `src/version.h` I think flags signal p2p _policies_ while `PROTOCOL_VERSION` _commits_ to which policy your node software is complying. Rust-bitcoin software should set their `PROTOCOL_VERSION` to which p2p policy they actually implement. As you said that would avoid receiving not-understood messages which may fail the connection.

Reply:
I don't think there is rational on how they're derived.

Reply:
I think the real confusion here is that the new PROTOCOL_VERSION has nothing to do with wtxidrelay - it really means "I support receiving support messages between version and verack". It just so happens that one such message is wtxidrelay.

Reply:
It is a bit unclear what version this library should set as it really depends on the functionality of the software using it. Would you be ok leaving it as it is and adding a comment similar to the one I've added?

Reply:
Right, I think maybe having a `PROTOCOL_VERSION` constant in the crate is a bit of a misnomer - originally `PROTOCOL_VERSION` implied messages and features supported, but these days it really only indicates breaking changes to the protocol (ie in this case sending messages between version and verack), whereas new features are introduced by simply sending new message types which old nodes will ignore. Its probably best to just add a list of protocol versions with detailed descriptions of what they imply, though I don't think we need to go back to 0, just add the few most recent ones.

Reply:
the downside I just realized for this constant here, is that you get `alert` messages from peers (because it's smaller than `70012`)

Reply:
Couldn't we (in a separate PR) remove the constant, use a hard-coded one for getheaders and getblocks and take the version for the version message as an argument from the user?